web-ng/src/app/components/layer-dialog/layer-dialog.component.ts

```diff
@@ -170,7 +170,7 @@ export class LayerDialogComponent implements OnDestroy {
     const forms = this.layerService.createForm(formId, fields);
     const layer = new Layer(
       this.layer?.id || '',
-      /* index */ this.layersSize,
+      /* index */ this.layerCount,
```

Comment:
Sorry I just thought of this now - what if number of layers has changed while the layer dialog is opened (e.g., a layer was added in another window/tab)? In that case this index will be out of date. Can we instead assign this when the layer is added back to the project? Sorry about for late comment!

Reply:
Good Point 💯